internal/metrics/metricsManager.go

```diff
@@ -486,3 +537,24 @@ func (m *MetricsManager) MonitoredClusterCount(monitoredClusters int64) {
 func bytesToGib(bytes int64) int64 {
 	return bytes / int64(units.GiB)
 }
+
+// []string{labelStatus, labelReleaseID, labelCached}
+func (m *MetricsManager) InstallerCacheGetReleaseOK(releaseId string, cacheHit bool) {
+	m.serviceLogicInstallerCacheGetRelease.WithLabelValues(labelValueOK, releaseId, fmt.Sprintf("%t", cacheHit)).Inc()
+}
+
+func (m *MetricsManager) InstallerCacheGetReleaseTimeout(releaseId string, cacheHit bool) {
+	m.serviceLogicInstallerCacheGetRelease.WithLabelValues(labelValueTimeout, releaseId, fmt.Sprintf("%t", cacheHit)).Inc()
+}
+
+func (m *MetricsManager) InstallerCacheGetReleaseError(releaseId string, cacheHit bool) {
+	m.serviceLogicInstallerCacheGetRelease.WithLabelValues(labelValueError, releaseId, fmt.Sprintf("%t", cacheHit)).Inc()
```

Comment:
What kind of values would `releaseId` get?

Reply:
This could have huge cardinality, can we maybe simplify it to major-minor? Would that provide enough info?

Reply:
Fair point. Yes, I will look into doing that. Yes I suppose that combined with logging an error we would provide enough of a clue to anyone monitoring the application. 

I will change this.

Reply:
I thought about an edge case here and wonder how we should handle (given cardinality concerns)

Obtaining a majorMinorVersion can fail

```
majorMinorVersion, err := ocRelease.GetMajorMinorVersion(i.log, releaseID, releaseIDMirror, pullSecret)
			if err != nil {
				i.log.Warnf("unable to get majorMinorVersion to record metric for %s falling back to full URI", releaseID)
				majorMinorVersion = releaseID
			}
```

In the snippet above, I go for what I feel is the most sensible option, to fall back to `releaseID` if this fails, what do you think?

Reply:
I think in this case we can add `unknown` as release and further info can be found on the warning message log. This will keep cardinality low even if we are not able to extract major-minor version. WDYT?

Reply:
Yes, good idea

Reply:
BTW I think extracting the version from the image could be costly, not sure it's worth it in this case. Is there any easy way we can have this value without this overhead?

Reply:
I don't think there are any guarantees about the image URI, it could be based on digest rather than tag for example, in these cases, we cannot infer the version from the URI.

My assumption is that there is no other reliable way to obtain this information, otherwise, why do we even need this function in the release? I can't see any solid alternatives in the code?

Reply:
> not sure it's worth it in this case

Another option is to remove the release label entirely.

Given that we are moving in a direction of only reporting timeout (based on the other comment we are still discussing) then it may be acceptable to simply say "there are timeouts - check it out" rather than labelling a version.

Further info could be gathered from logs anyway.

WDYT?

Reply:
That's a good point, we probably should use this value. Could this value be somehow recycled? Are we using it already up the chain? Are we saving it in the DB? Do we have it accessible in this scope? Can it be made accessible easily?

Reply:
>If we have only the full OpenshiftVersion value, we could extract major-minor from that with semver package

I think this may be the way to go, but beware that Openshift version is not technically the release version.

So we might be able to say "for this openshift version I had a problem with a release", that's about as truthful as we can be.

Reply:
As discussed I think you're right here and we could use OC based extraction if we don't find the value up the chain already

Reply:
Nothing found further up the chain, so will mark this one as resolved.